cmake/DaemonFlags.cmake

```diff
@@ -255,7 +314,7 @@ else()
 		endif()
 	endif()
 
-	if (USE_RECOMMENDED_CXX_STANDARD)
+	if (USE_RECOMMENDED_CXX_STANDARD AND NOT USE_CPP23)
```

Comment:
This doesn't seem right since if CPP23 is enabled then the standard flag is globally disabled, but the C++23 flag is only applied to particular modules.

I guess the C++23 per-module is what causes issues with PCH too? Since the PCH code has to manually flags from every source where flags can be injected so it may not be taking that into account.

Why not add the C++23 flag here globally in the same way that we add the C++14 flag? That could solve both the PCH problem and avoid needing a higher CMake version.